### PR TITLE
fine_grain_logger: make sure fine-grain-logger map registration in macros

### PR DESCRIPTION
**Pull request by botengyao**
The `ENVOY_LOG_COMP_LEVEL` will lead to non-active fine-grain loggers are not managed by the fine-grain logger map when log level is lower than default, since `ENVOY_LOG_TO_LOGGER` is the only way to call `FINE_GRAIN_LOG` to initialize a fg-logger.

This PR creates a new `FINE_GRAIN_LOGGER()` and that can be used for getting fine-grain logger. 
Also creates a new `ENVOY_LOG_COMP_LEVEL_FINE_GRAIN_IF()` to consider fine-grain logger when doing comparison. 

Note, not all marcos are migrated.

Commit Message:
Additional Description:
Risk Level: low
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by botengyao** [comment]
Hi @KBaichoo, this is ready for a review. Thanks!

**Review by KBaichoo — APPROVED** [review]
lgtm apart from minor change

File: source/common/common/fine_grain_logger.h [hunk 1]
```diff
@@ -124,6 +124,14 @@ class FineGrainLogContext {
    */
   static bool safeFileNameMatch(absl::string_view pattern, absl::string_view str);
 
+  /**
+   * Remove a fine grain log entry for testing only.
+   */
+  void removeFineGrainLogEntry(absl::string_view key) ABSL_LOCKS_EXCLUDED(fine_grain_log_lock_) {
```
[the remark this answers is not in the document]
Comment (botengyao) [reply source/common/common/fine_grain_logger.h]: done, thanks!

**Review by KBaichoo — APPROVED** [review]

**Review by ggreenway — APPROVED** [review]

**Pull request merged**